### ISSUE TITLE
Animation is jittery/laggy

### ISSUE THREAD

**Issue by ReidLitkauo** [ReidLitkauo/place#1]
Especially noticeable on Firefox, but the lag is noticeable on Chrome too.

Zooming in too much on Firefox results in discrete movement steps being visible when dragging the canvas about. The selection reticle lags behind the canvas a lot, making it fall out of sync with the canvas when moving about.

**Comment by ReidLitkauo** [comment]
The jittery movement was caused by using too large of numbers when positioning the canvas -- the limit of floating-point precision was reached, and the gaps between numbers displayed as gaps hindering smooth animation/positioning. I fixed this by moving the `MainCanvas` event handling hooks from the canvas itself to its parent, which let me drop the massive border and allowed me to position the canvas without accounting for all that space. This led to the system using smaller numbers, and allowed for more fine-grained animation/positioning.

The reticle is still a frame or so behind the canvas animation, which I believe is caused by the reticle animating first in every frame; it uses canvas positioning data that's out-of-date by the time the canvas animates in that same frame. I could probably solve this by moving the reticle's animation logic to the MainCanvas, but that seems pretty messy. Perhaps I could introduce some sort of parameter in `AnimatedComponent`'s constructor that will specify each element's position in the animation frame... For now, I'm just gonna leave it as-is. The lagging isn't that bad, and it does give the reticle an interesting "bouncing" effect.
